students/srmorehouse/Lesson09/donor_models.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+import operator
+import os
+import sys
+
+"""
+Steve Morehouse
+Lesson 09
+"""
+
+class Donor (object):
+
+    def __init__(self, name, donations=None):
+        self.name = name
+        self.donations = donations if donations is not None else []
+
+    
+    def add_donation (self, amount):
+        self.donations.append(amount)
+    
+    
+    def compose_thank_you (self):
+        msg = f"\n{self.name},\n\nThank you for your donation of ${self.donations[-1]:.2f}.\n"
+        return msg
+    
+    
+    def get_donor_summary(self):
+        num_donations = len(self.donations)
+        sum_donations = sum(self.donations)
+        average_donation = sum_donations/num_donations if num_donations else 0
+
+        return (self.name, sum_donations, num_donations, average_donation)
+    
+    
+class DonorCollection():
+
+    def __init__(self):
+        self.donors={}
+
+ 
+    def add (self, donor):
+        self.donors[donor.name] = donor
+    
+    
+    def list_donors (self):
+        return_list = ''
+        for donor in self.donors:
+            return_list += f'{donor}\n'
+        return return_list
+
+
+    def donor_exists(self, name):
+        return any(d for d in self.donors.values() if d.name.upper() == name.upper())
```

Comment:
since you're storing donor data as a dictionary, you should be able to easily access it via dictionary key:

```
return bool(self.donors.get(name))

```

if you're worried about string case then I would recommend storing it in all lower case and then adjust on display